test/manipulation.test.js

```diff
@@ -57,6 +57,20 @@ describe('manipulation', function () {
       });
     });
 
+    it('should create instance (promise variant)', function (done) {
+      Person.create({name: 'Anatoliy'})
+      .then (function (p) {
```

Comment:
I can change it, but let me defend this style first... :)

Firstly, .then is a chained function from the previous statement that
returned a promise in the same context, same as .catch.  There can be
multiple .thens chained together, and I like the indentation to actually
mean a scope change...

https://www.dropbox.com/s/6ofmy3jx6i6pyku/Screenshot%202015-02-19%2014.24.53.png?dl=0
https://www.dropbox.com/s/mgf4k5oiob9hx1g/Screenshot%202015-02-19%2014.26.01.png?dl=0

Does the extra indentation in the second version make it any clearer?  I
prefer the first version where each indentation is clearly a change in
context.

Taken to extremes, indenting chained .thens can actually reduce
readability... Here's one of the tests in relations.test.js, a common
promise pattern:
https://www.dropbox.com/s/xdijmj9340szrdv/Screenshot%202015-02-19%2014.40.54.png?dl=0

and here it is with indented .thens:
https://www.dropbox.com/s/b3muldxq44k7xh8/Screenshot%202015-02-19%2014.42.16.png?dl=0

...that's clearly ridiculous, so maybe just indent the first .then
https://www.dropbox.com/s/7iqqsda2gd1w8hg/Screenshot%202015-02-19%2014.51.53.png?dl=0

Better, but now the indentation rule seems more arbitrary and subjective to
me... I can live with it, though, if you insist :)

Secondly, the space after .then is my mistake,  The space after function is
not... if there is no space, it looks more like it is calling a function
rather than defining it (my editor actually highlights them differently).

https://www.dropbox.com/s/zkw2jwywyxeky6p/Screenshot%202015-02-19%2014.10.36.png?dl=0
https://www.dropbox.com/s/zm1mzh00zk28e2h/Screenshot%202015-02-19%2014.11.58.png?dl=0

...also I notice that *most* other places in the file, the callbacks are
defined with a space after function (e.g.: function (done) 2 lines above)

On Thu, Feb 19, 2015 at 10:05 AM, Miroslav Bajtoš <notifications@github.com>
wrote:

> In test/manipulation.test.js
> <https://github.com/strongloop/loopback-datasource-juggler/pull/451#discussion_r25004746>
> :
>
> > @@ -57,6 +57,20 @@ describe('manipulation', function () {
> >        });
> >      });
> >
> > +    it('should create instance (promise variant)', function (done) {
> > +      Person.create({name: 'Anatoliy'})
> > +      .then (function (p) {
>
> Coding style: indent .then by two spaces, no space between opening brace.
>
> Person.create(...)
>   .then(function(p) {
>      // ...
>   })
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/loopback-datasource-juggler/pull/451/files#r25004746>
> .
>